client/network/src/service.rs

```diff
@@ -889,16 +891,30 @@ impl Metrics {
 			connections_closed_total: register(CounterVec::new(
 				Opts::new(
 					"sub_libp2p_connections_closed_total",
-					"Total number of connections closed, by direction, reason and by being the first or not"
+					"Total number of connections closed, by direction and reason"
 				),
-				&["direction", "reason", "was_first"]
+				&["direction", "reason"]
 			)?, registry)?,
 			connections_opened_total: register(CounterVec::new(
 				Opts::new(
 					"sub_libp2p_connections_opened_total",
-					"Total number of connections opened by direction and by being the first or not"
+					"Total number of connections opened by direction"
 				),
-				&["direction", "is_first"]
+				&["direction"]
+			)?, registry)?,
+			distinct_peers_connections_closed_total: register(CounterVec::new(
+				Opts::new(
+					"sub_libp2p_distinct_peers_connections_closed_total",
+					"Total number of connections closed with distinct peers, by direction and reason"
+				),
+				&["direction", "reason"]
```

Comment:
```suggestion
```

I didn't think about it, but I don't think it makes sense to have the `direction` or `reason` on these metrics.

It would be misleading to report a single `direction` while we might have multiple connections with multiple directions.

Similarly, the `reason` here is the reason for closing the last connection, which to me makes the last connection special while it shouldn't be.
For instance, if a severe protocol error happens on the "main" connection, we immediately close it and also close all the other connections to that same peer afterwards. The `reason` reported here would be the one of the other connection (if any), which doesn't really make sense.